crates/rpc-types-eth/src/transaction/mod.rs

```diff
@@ -259,6 +265,108 @@ impl<T: TransactionTrait + Encodable2718> TransactionResponse for Transaction<T>
     }
 }
 
+#[cfg(feature = "serde")]
+mod tx_serde {
+    //! Helper module for serializing and deserializing OP [`Transaction`].
+    //!
+    //! This is needed because we might need to deserialize the `gasPrice` field into both
+    //! [`crate::Transaction::effective_gas_price`] and [`alloy_consensus::TxLegacy::gas_price`].
+    use super::*;
+    use serde::{Deserialize, Serialize};
+
+    /// Helper struct which will be flattened into the transaction and will only contain `gasPrice`
+    /// field if inner [`TxEnvelope`] did not consume it.
```

Comment:
is this a serde flatten hack?

Reply:
yep, it relies on it being deserialized after the inner tx. if it was just a field it would always be deserialized and consumed before